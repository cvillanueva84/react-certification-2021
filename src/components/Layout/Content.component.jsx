import React from 'react';

import ContentWrapper from './Content.styles';
import VideoComponent from './Video.component';

const data = require('../../mockdata/youtube-videos-mock.json');

function ContentComponent() {
    const element = data && data.items.map((item)=>(
      <VideoComponent description={item.snippet.description} key={item.etag} title={item.snippet.title} url={item.snippet.thumbnails.medium.url} />
    ))
    return (
      <ContentWrapper>
        {element}
      </ContentWrapper>
    )
}

export default ContentComponent;