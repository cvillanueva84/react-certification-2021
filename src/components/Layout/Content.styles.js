import styled from 'styled-components';

const ContentWrapper = styled.div`
    display: flex;
    justify-content: space-between;
    flex-wrap: wrap;
    margin: 0 auto;
    width: 90%;
    row-gap: 10px;
    column-gap: 2em;
`;

export default ContentWrapper;