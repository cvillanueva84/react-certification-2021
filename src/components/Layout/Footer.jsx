import React from 'react';

import styled from 'styled-components';

const P = styled.p`
  color: grey;
  font-size: 1.1em;
`;

const Small = styled.p`
  color: grey;
  font-size: 0.5em;
`;

const Footer = () => {
  return (
    <footer className="footer mt-auto py-3 bg-dark">
      <div className="container text-center">
        <P className="text-muted text-center">Luis F. Infanz√≥n M.</P>
        <Small className="text-muted text-center">
          Wizeline Academy - 2021 React Bootcamp
        </Small>
      </div>
    </footer>
  );
};

export default Footer;
