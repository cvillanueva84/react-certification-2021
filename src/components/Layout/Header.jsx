import React from 'react';

import styled from 'styled-components';

import { Link } from 'react-router-dom';

import Buscar from './Buscar';
import Navegacion from './Navegacion';

const Nav = styled.nav`
  background-color: #8ec045;
`;

const Header = () => {
  return (
    <Nav className="navbar navbar-expand-lg navbar-light">
      <div className="container-fluid">
        <Link to="/" className="navbar-brand text-uppercase">
          YouTube client app
        </Link>
        <button
          className="navbar-toggler"
          type="button"
          data-bs-toggle="collapse"
          data-bs-target="#navbarText"
          aria-controls="navbarText"
          aria-expanded="false"
          aria-label="Toggle navigation"
        >
          <span className="navbar-toggler-icon" />
        </button>
        <div className="collapse navbar-collapse" id="navbarText">
          <Navegacion />
          <span className="navbar-text">
            <Buscar />
          </span>
        </div>
      </div>
    </Nav>
  );
};

export default Header;
