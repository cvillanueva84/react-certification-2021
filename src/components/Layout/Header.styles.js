import styled from 'styled-components';

export const HeaderWrapper = styled.div`
  align-items: center; 
  background-color: #f0575d;
  border-bottom: 1px solid rgba(0, 0, 0, 0.0975);
  color: "#FFFFFF";
  display: flex;
  justify-content: space-between;
  margin: 0 auto;
  padding: 10px 20px;
  width: 100%;

  p{
    color: #FFFFFF;
  }
`;
 
export const LogoLink = styled.div`
    margin: 0px 20px;
    img {
        max-width: 50px;
        height: auto;
    }
`;
 
export const MenuToggle = styled.div`
    display: block;
    width: 50px;
    padding: 5px;
    margin: 0px 20px;
    svg{
        height: auto;
        max-width: 100%;
    }
`;

export const ButtonToggle = styled.div`
    display: block;
    margin-left: auto;
    width: 30px;
    margin-right: 10px;
    svg{
        height: auto;
        max-width: 100%;
    }
`;
 
export const Input = styled.input`
  font-size: 16px;
  border: solid 1px #dbdbdb;
  border-radius: 3px;
  color: #262626;
  padding: 7px 33px;
  border-radius: 3px;
  color: #999;
  cursor: text;
  font-size: 14px;
  font-weight: 300;
  text-align: center;
  background: #fafafa;
 
  &:active,
  &:focus {
    text-align: left;
  }
`;
 
export const MenuLink = styled.a``;