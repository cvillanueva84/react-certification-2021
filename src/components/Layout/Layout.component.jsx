import React from 'react';
import styled from 'styled-components';

const LayoutStyled = styled.main`
  width: 100%;
  display: flex;
  flex-direction: column;
  padding: 0;
  box-sizing: border-box;
`;

function Layout({ children }) {
  return <LayoutStyled>{children}</LayoutStyled>;
}

export default Layout;
