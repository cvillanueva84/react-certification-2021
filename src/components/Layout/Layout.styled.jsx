import styled from 'styled-components';

export const Container = styled.main`
  color: ${(props) => props.theme['text-color']};
  background-color: ${(props) => props.theme['background-main']};
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;

export const InputSearch = styled.input`
  color: white;
  border-radius: 4px;
  border: 0;
  height: 1.1876em;
  margin: 0;
  display: block;
  padding: 10px 7px 8px 48px;
  min-width: 0;
  background: none;
  box-sizing: content-box;
  &:focus {
    border: 0;
    outline: none;
  }
`;
