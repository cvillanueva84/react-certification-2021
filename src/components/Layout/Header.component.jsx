import React from 'react';

import { faBars, faToggleOff } from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { ButtonToggle, HeaderWrapper, Input, LogoLink, MenuToggle } from './Header.styles';
import SessionOutLogo from '../../resources/session_out.png';

function HeaderComponent() {
  return (
    <HeaderWrapper>
      <Input type='text' placeholder="Wizeline" />
      <ButtonToggle>
        <FontAwesomeIcon icon={faToggleOff} size="6x" style={{ color: 'white'}} title="toggle-button" />
      </ButtonToggle>
      <p>Dark Mode</p>
      <LogoLink href="#">
        <img src={SessionOutLogo} alt="Logo" />
      </LogoLink>
      <MenuToggle href="#">
        <FontAwesomeIcon icon={faBars} size="6x" style={{ color: 'white'}} title="menu-toggle" />
      </MenuToggle>
    </HeaderWrapper>
  );
}

export default HeaderComponent;
