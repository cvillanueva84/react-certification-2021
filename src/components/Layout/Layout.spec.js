import Layout from './Layout.component';

const { render, screen } = require('@testing-library/react');

describe('Layout', () => {
  test('Should renders Layout component', () => {
    const { container } = render(<Layout />);
    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test('Should contains a "main" tag', () => {
    const { container } = render(<Layout />);
    const mainElement = container.querySelector('main');
    expect(mainElement).toBeInTheDocument();
  });

  test('Should renders a children element', () => {
    const { container } = render(
      <Layout>
        <p>Hello world!</p>
      </Layout>
    );
    const children = screen.getByText(/Hello world!/i);
    expect(children).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
