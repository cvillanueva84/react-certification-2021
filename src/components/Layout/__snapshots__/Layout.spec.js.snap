// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Should renders Layout component 1`] = `
<div>
  <main
    class="sc-bdnxRM iYBsxF"
  />
</div>
`;

exports[`Layout Should renders a children element 1`] = `
<div>
  <main
    class="sc-bdnxRM iYBsxF"
  >
    <p>
      Hello world!
    </p>
  </main>
</div>
`;
